src/clients/python/library/tritonclient/utils/cuda_shared_memory/__init__.py

```diff
@@ -176,10 +176,17 @@ def set_shared_memory_region(cuda_shm_handle, input_values):
     offset_current = 0
     for input_value in input_values:
         input_value = np.ascontiguousarray(input_value).flatten()
-        byte_size = input_value.size * input_value.itemsize
-        _raise_if_error(
-            c_int(_ccudashm_shared_memory_region_set(cuda_shm_handle, c_uint64(offset_current), \
-                c_uint64(byte_size), input_value.ctypes.data_as(c_void_p))))
+        if input_value.dtype == np.object_:
+            input_value = input_value.item()
+            byte_size = np.dtype(np.byte).itemsize * len(input_value)
```

Comment:
@deadeyegoodwin / @tanmayv25 Is it ok if I pass the pointer to a temporary variable? I was not sure about the valid duration of a pointer in python.

Reply:
I assume you are talking about 'input_value'. You do have to be careful that the python object lifetime is longer than any use you have in C code. But in this case it seems that you haven't changed that compared to what was before, correct? Did you change _ccudashm_shared_memory_region_set?

Reply:
No, I didn't change the `_ccudashm_shared_memory_region_set`. I just used `input_value.item()` instead.

Reply:
Then should be ok. I believe that function just copies data out of input_value.item and into shared memory. So it doesn't keep reference to item() pointer.

Reply:
I see.